src/depackers/vorbis.c

```diff
@@ -1195,7 +1192,7 @@ static int STBV_CDECL uint32_compare(const void *p, const void *q)
 
 static int include_in_sort(Codebook *c, uint8 len)
 {
-   if (c->sparse) { return (len != NO_CODE); }
+   if (c->sparse) { assert(len != NO_CODE); return TRUE; }
```

Comment:
1) Usage of `include_in_sort` at line 1216 is irrelevant to sparse codebooks.
2) Usage of `include_in_sort` later in `compute_sorted_huffman` with `huff_len` is relevant for all codebooks. (line 1235)
3) `huff_len` = `lengths[values[i]]` for sparse codebooks. (line 1234)
4) 0 <= `i` < `c->sorted_entries`. (lines 1227-1233)
5) `compute_sorted_huffman` is called from `start_decoder`. (line 3900)
6) `values` is initialized in `start_decoder` via `compute_codewords` (line 3886)
7) These are initialized by `add_entry`, which for sparse codebooks, inserts them incrementally and only for non-`NO_CODE` lengths. The value will always be an index to that length in the lengths table. (lines 1124, 1147)
8) One value is inserted into values in `compute_codewords` for every non-`NO_CODE` length.
9) `c->sorted_entries` is always equal to the number of non-`NO_CODE` lengths inserted in sparse codebooks (see previous comment).

`len` will never be `NO_CODE` in `include_in_sort` for sparse codebooks, so line 1198 will always return `TRUE`.